pkg/repometrics/stats.go

```diff
@@ -37,35 +113,80 @@ type Contributor struct {
 	Commits int    `json:"commits"`
 }
 
+// Relase represents a release from a repository
+type Release struct {
+	AssetsURL   string    `json:"assets_url,omitempty"`   // URL of the release assets.
+	Date        time.Time `json:"date"`                   // Date of the release.
+	Name        string    `json:"name"`                   // Name of the release.
+	TarballURL  string    `json:"tarball_url,omitempty"`  // URL of the release tarball.
+	VulnRefs    ExtRefs   `json:"vuln_refs,omitempty"`    // List of references to SBOM, VEX and Vulnerabilities.
+	MissingRefs ExtRefs   `json:"missing_refs,omitempty"` // List of missing references to SBOM, VEX and Vulnerabilities.
+	ReleaseURL  string    `json:"release_url,omitempty"`  // URL of the release.
+	UploadURL   string    `json:"uploaded,omitempty"`     // URL of the release upload.
+}
+
+// ExtRefs is information pointing to the SBOM, VEX and Vulnerabilities files that exists.
+type ExtRefs struct {
+	SBOMCDX string `json:"sbomCDX"`
+	VEXCDX  string `json:"vexCDX"`
+	VULNCDX string `json:"vulnerabilityCDX"`
+	VULNMD  string `json:"vulnerabilityMD"`
+	Latest  bool   `json:"actual"`
+}
+
+// CVE is an assembly of the Vulnerability, Exploitability and SBOMs
+type CVE struct {
+	Vulnerability  vul.ID   `json:"id"`             // the Vulnerability
+	Exploitability string   `json:"exploitability"` // the Exploitability not correct type yet
+	SBOMs          []string `json:"sboms"`          // the SBOMs not correct type yet
+}
+
+// NewStats resets the statistics
 func NewStats() *Stats {
 	return &Stats{
 		TopCommitters:   make([]Contributor, 0),
 		TopCommitters1Y: make([]Contributor, 0),
 	}
 }
 
+// sorts the contributors
 func sortContributors(contributors []Contributor) {
 	sort.Slice(contributors, func(i, j int) bool {
 		return contributors[i].Commits > contributors[j].Commits
 	})
 }
 
-func getTopContributors(contributors []Contributor, limit int) []Contributor {
+// find the top contributors
+func getTopContributors(contributors []Contributor, limit int) []Contributor { //nolint:unparam
+	sortContributors(contributors)
 	if len(contributors) > limit {
 		return contributors[:limit]
 	}
 	return contributors
 }
 
+// guess whether a contributor is a bot or not
 func isBot(login string) bool {
-	return strings.Contains(login, "[bot]")
+	return strings.Contains(login, "[bot]") ||
```

Comment:
The list of bots could be put into a variable:

```go
botPatterns := []string{"bot", "action", "istio-testing"}
_, found := xslices.FindFunc(botPatterns, func(p string) { return strings.Contains(login, p) })
return found
```

Note that if the pattern includes "bot", then you don't have to include "dependabot", "renovate-bot", "[bot]" or anything else with "bot" in it.